src/pygeosx/pysolver/PySolver.cpp

```diff
@@ -0,0 +1,303 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+// Source includes
+#include "../pygeosx.hpp"
+#include "PySolver.hpp"
+#include "../PyGroup.hpp"
+
+#define VERIFY_NON_NULL_SELF( self ) \
+  PYTHON_ERROR_IF( self == nullptr, PyExc_RuntimeError, "Passed a nullptr as self.", nullptr )
+
+#define VERIFY_INITIALIZED( self ) \
+  PYTHON_ERROR_IF( self->group == nullptr, PyExc_RuntimeError, "The PySolver is not initialized.", nullptr )
+
+
+namespace geosx
+{
+extern std::unique_ptr< GeosxState > g_state;
+
+extern bool g_alreadyInitialized;
+
+namespace python
+{
+
+struct PySolver
+{
+  PyObject_HEAD
+
+  static constexpr char const * docString =
+    "A Python interface to geosx::SolverBase.";
+
+  geosx::SolverBase * group;
+  geosx::ProblemManager * pb_manager;
```

Comment:
What does adding the `ProblemManager` here help? I'd imagine it might be better to just access everything through `GeosxState`.

Reply:
I rathered having a pointer to ProblemManager as I didn't want to go through GeosxState each time, I don't really remember why though...

Reply:
Done.